src/lib-components/lists/DetailList.vue

```diff
@@ -1,45 +1,44 @@
 <script setup lang="ts">
+import { computed, ref, watch } from "vue"
 import { useAppFlasher } from "@/composables"
-import { ref, watch } from "vue"
+import { DetailListConfig, DetailListAPI } from "@/composables/list"
 import BaseAPI from "../../api/base"
 import DateFilter from "../layout/DateFilter.vue"
 import Paginator from "../navigation/Paginator.vue"
 
-const props = withDefaults(
-  defineProps<{
-    refreshTrigger?: number
-    reloadTrigger?: number
-    title: string
-    url: string
-  }>(),
-  {
-    refreshTrigger: 0,
-    reloadTrigger: 0,
-  }
-)
+const props = defineProps<{
+  config: DetailListConfig
+}>()
 
 const dateRange = ref<{ minDate: number; maxDate: number }>({
   minDate: 0,
   maxDate: 0,
 })
-const hasContent = ref(true)
+
 const items = ref<any[]>([])
+
 const pagination = ref({
   page: 1,
-  perPage: 10,
+  perPage: props.config.perPage || 10,
```

Comment:
After flattening the props we can leverage `withDefaults` for setting these types of default values and limit the number of potentially `undefined` props.